src/Discord/Discord.php

```diff
@@ -786,34 +785,36 @@ protected function handleDispatch(object $data): void
                 $this->emit($alternative, [$d, $this]);
             }
 
-            if ($data->t == Event::MESSAGE_CREATE && mentioned($this->client->user, $new)) {
+            if ($data->t === Event::MESSAGE_CREATE && mentioned($this->client->user, $new)) {
                 $this->emit('mention', [$new, $this, $old]);
             }
-        }, function ($e) use ($data) {
+        };
+
+        $onReject = function ($e) use ($data) {
             if ($e instanceof \Error) {
                 throw $e;
-            } elseif ($e instanceof \Exception) {
+            }
+
+            if ($e instanceof \Exception) {
                 $this->logger->error('exception while trying to handle dispatch packet', ['packet' => $data->t, 'exception' => $e]);
             } else {
                 $this->logger->warning('rejection while trying to handle dispatch packet', ['packet' => $data->t, 'rejection' => $e]);
             }
-        });
+        };
 
         $parse = [
             Event::GUILD_CREATE,
             Event::GUILD_DELETE,
         ];
-
-        if (! $this->emittedInit && (! in_array($data->t, $parse))) {
-            $this->unparsedPackets[] = function () use (&$handler, &$deferred, &$data) {
-                /** @var ExtendedPromiseInterface */
+        
+        if (! $this->emittedInit && ! in_array($data->t, $parse)) {
+            $this->unparsedPackets[] = function () use (&$handler, &$onResolve, &$onReject, &$data) {
```

Comment:
Why are you passing these as reference?

Reply:
&$deferred was previous passed by reference